model-optimizer/extensions/middle/EltwiseInputReshape.py

```diff
@@ -64,59 +63,86 @@ def find_and_replace_pattern(self, graph: Graph):
-                    node_copy = Op.create_input_data_node(graph, node.id + '/copy', value=np.array(new_value))
-                    for consumer in mapping[shape_key]:
-                        edge_attrs = graph.get_edge_data(node.id, consumer.id)[0]
-                        del edge_attrs['new_shape']
-
-                        # Remove edge from previous data node and connect new data node with its consumer
-                        graph.remove_edge(node.id, consumer.id)
-                        graph.add_edge(node_copy.id, consumer.id, **edge_attrs)
+# compute a map of unsqueeze_dims for each producer of eltwise node
+def compute_unsqueeze_map_for_eltwise(eltwise_node: Node):
+    eltwise_shape = eltwise_node.out_port(0).data.get_shape()
+    max_dims = None
+    for in_port_idx in eltwise_node.in_ports():
+        consumer_port = eltwise_node.in_port(in_port_idx)
+        producer_port = consumer_port.get_source()
+        producer_shape = producer_port.data.get_shape()
+        if max_dims is None or len(producer_shape) > max_dims:
+            max_dims = len(producer_shape)
+
+    axis = eltwise_node.axis if eltwise_node.has_valid('axis') else None
+    unsqueeze_dims_map = {}
+    for in_port_idx in eltwise_node.in_ports():
+        consumer_port = eltwise_node.in_port(in_port_idx)
+        producer_port = consumer_port.get_source()
+        producer_shape = producer_port.data.get_shape()
+
+        unsqueeze_dims = int64_array([])
+        # 1. Compute unsqueeze dimensions in the tail
+        if len(producer_shape) != max_dims and len(producer_shape) > 0 and axis is not None:
+            num_unsqueeze_dims = max_dims - axis - len(producer_shape)
+            if num_unsqueeze_dims > 0:
+                unsqueeze_dims = np.arange(len(producer_shape), len(producer_shape) + num_unsqueeze_dims,
+                                           dtype=np.int64)
+
+        # 2. Compute unsqueeze dimensions in the head
+        unsqueeze_dims_head = np.arange(len(eltwise_shape) - len(producer_shape) - len(unsqueeze_dims), dtype=np.int64)
+
+        # Pay attention that unsqueeze dims order makes sense
+        # since shape is normalized in the tail first and after in the head
+        unsqueeze_dims = np.concatenate((unsqueeze_dims, unsqueeze_dims_head))
+        unsqueeze_dims_map[producer_port] = unsqueeze_dims
+
+    return unsqueeze_dims_map
+
+
+# normalize input shapes for eltwise node by using Unsqueeze operation
+def normalize_eltwise_inputs(graph: Graph):
+    # Generate a map for producers of eltwise nodes with non-normalized shapes
+    # and in this map every producer has another map that reflects normalized shape
+    # to a list of eltwise consumers
+    mapping = {}
+    for eltwise_node in graph.get_op_nodes(is_eltwise=True):
+        unsqueeze_dims_map = compute_unsqueeze_map_for_eltwise(eltwise_node)
+        for in_port_idx in eltwise_node.in_ports():
+            consumer_port = eltwise_node.in_port(in_port_idx)
```

Comment:
`for in_port_idx, consumer_port in eltwise_node.in_ports().items():`

Reply:
Done.